SzczepionkaWEB/src/app/component/register/register.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, OnInit } from '@angular/core';
+import {FormBuilder, FormGroup, Validators} from '@angular/forms';
+import {ActivatedRoute, Router} from '@angular/router';
+import {RegisterService} from '../../service/register/register.service';
+import {Registration} from '../../model/Registration';
+
+@Component({
+  selector: 'app-register',
+  templateUrl: './register.component.html',
+  styleUrls: ['./register.component.css']
+})
+export class RegisterComponent implements OnInit {
+  registerForm!: FormGroup;
```

Comment:
What does actually this '!' mean? Is it a typescript sign for a neccessary field?

Reply:
This is something like "this expression cannot be null or undefined". It's  set to avoid problems with compiler. (Don't  know why, but without this there were information that registerForm may be null or undefined. 

By this exclamation mark, we want to say to compiler "I know that it will be never null so don't cry about this.". :D